### PR TITLE
Updated Wwise detection to account for Godot plugin libraries and new Unity library

### PR DESCRIPTION
**Pull request by MonstoBusta**
### SteamDB app page links to a few games using this

Runa & the Chaikurú Legacy: https://steamdb.info/app/2283470/ (depot showing the library in the files: https://steamdb.info/depot/2283472/)

### Brief explanation of the change
Godot uses an unofficial but semi officially endorsed plugin to add Wwise to its games. However, the plugin's library file is different from the standard AkSoundEngine.dll. I have added the Godot version for both Godot 3.x and Godot 4.x to rules.ini and the test types folder. I did not account for WaapiGDNative.dll/so, but WWiseGDNative.dll/so should be enough. I don't know if this works so I would like someone with more knowledge of regex to see if what I typed works.

### CODE REVIEW

**Comment by SteamDB-Tracker** [comment]
After running changed rules in this pull request *(this bot only tests rules.ini changes)*:

Changed technologies: **SDK.Wwise**

**2012** apps still match.

These apps now match: **(13)** [1707220](https://steamdb.info/app/1707220/depots/) [1786010](https://steamdb.info/app/1786010/depots/) [1965810](https://steamdb.info/app/1965810/depots/) [2065530](https://steamdb.info/app/2065530/depots/) [2118940](https://steamdb.info/app/2118940/depots/) [2283470](https://steamdb.info/app/2283470/depots/) [2407540](https://steamdb.info/app/2407540/depots/) [2422440](https://steamdb.info/app/2422440/depots/) [2429640](https://steamdb.info/app/2429640/depots/) [2512930](https://steamdb.info/app/2512930/depots/) [3179920](https://steamdb.info/app/3179920/depots/) [3227630](https://steamdb.info/app/3227630/depots/) [3240600](https://steamdb.info/app/3240600/depots/)

Please confirm that all these changed apps are correct.

File: tests/types/SDK.Wwise.txt [hunk 1]
```diff
@@ -10,3 +10,15 @@ DeadByDaylight/Plugins/Wwise/ThirdParty/x64_vc160/Release/bin/AkSoundEngineDLL.d
...
+libWwiseGDNative.so
+/libWwiseGDNative.so
+/libwwise.windows.template_release.dll
+/libwwise.windows.template_debug.dll
+libwwise.windows.template_release.dll
+libwwise.windows.template_debug.dll
+/libwwise.linux.template_release.so
+/libwwise.linux.template_debug.so
+libwwise.linux.template_release.so
+libwwise.linux.template_debug.so
```
Comment (Lifeismana) [inline tests/types/SDK.Wwise.txt:24]:
```suggestion
WwiseGDNative.dll
/WwiseGDNative.dll
build/windows/WwiseGDNative.dll
```

**Comment by MonstoBusta** [comment]
I can confirm that these changed apps all use Wwise.

File: tests/types/_NonMatchingTests.txt [hunk 2]
```diff
@@ -63,6 +63,10 @@ argss102j.dll
 argss102e.dll
 Sub/Folder/ARGSS104E.dll
 CgS_2019.dll
+libwwise.windows.debug.dll
+libwwise.windows.release.dll
+libwwise.linux.release.so
+libwwise.linux.debug.so
```
Comment (Lifeismana) [inline tests/types/_NonMatchingTests.txt:69]: since we're only keeping the Godot 3 for now, can you replace these non matching test with ones for godot 3 and move them next to the existing ones for that rule (AkSoundEngine.dll)

File: rules.ini [hunk 3]
```diff
@@ -353,7 +353,9 @@ V8 = (?:^|/)v8\.dll$
 Vivox[] = (?:^|/)ortp\.dll$
 Vivox[] = vivox
 Vorbis = vorbis
-Wwise = (?:^|/)AkSoundEngine(?:dll)?\.(?:bundle|dll)$
+Wwise[] = (?:^|/)AkSoundEngine(?:dll)?\.(?:bundle|dll)$
+Wwise[] = (?:^|/)(?:lib)?WwiseGDNative\.(?:dll|so)$
+Wwise[] = (?:^|/)libwwise\.(?:windows|linux)\.template(?:_release|_debug)\.(?:dll|so)$
```
Comment (Lifeismana) [inline rules.ini:358]:
```suggestion
Wwise[] = (?:^|/)AkSoundEngine(?:dll)?\.(?:bundle|dll)$
Wwise[] = (?:^|/)WwiseGDNative\.dll$
```

**Review by Lifeismana — CHANGES_REQUESTED** [review]
There's only 1 app that's using (based on what you said) Godot 4 & wwise that's not really enough to edit that rule to match these for now
Let's only keep the godot 3 version

**Comment by MonstoBusta** [comment]
I would also like to say that 2118940 is a Godot game (Godot 3.x), as is 3179920 (Godot 4.x), since to my knowledge only the Godot wwise plugins have the libraries named like that

**Comment by MonstoBusta** [comment]
> There's only 1 app that's using (based on what you said) Godot 4 & wwise that's not really enough to edit that rule to match these for now Let's only keep the godot 3 version

How many Godot 4.x games using Wwise must there be in order for those to be matched? The file for the 4.x plugin is from the same plugin as the 3.x extension, only updated. This plugin has also been put in a directory made by the Godot Foundation (https://github.com/godot-sdk-integrations).

**Comment by Lifeismana** [comment]
5 to 10 apps (that are unique (demo & game itself count as 1), with 10 being preferable to not have too many edge case

**Comment by MonstoBusta** [comment]
> 5 to 10 apps (that are unique (demo & game itself count as 1), with 10 being preferable to not have too many edge case

So I presume that Chicken Fall Prologue is being counted as a separate game here rather than a demo (with Chicken Fall and Chicken Fall Prologue both having demos). Fine, I'll accept.

File: tests/types/SDK.Wwise.txt (hunk 1 above)
Comment (MonstoBusta) [reply tests/types/SDK.Wwise.txt:24]: Why not include Linux libraries as well?

Comment (MonstoBusta) [reply tests/types/SDK.Wwise.txt:24]: > 

The projects include Linux versions, so why not account for those as well?

**Review by MonstoBusta — COMMENTED** [review]
I don't appreciate you removing Linux suport, but whatever

**Comment by Lifeismana** [comment]
There's no use in keeping the linux version when it doesn't help to detect more games

**Comment by MonstoBusta** [comment]
apologies, I closed the pull request hoping that it wouldn't put my Unity commits into here

**Comment by SteamDB-Tracker** [comment]
After running changed rules in this pull request *(this bot only tests rules.ini changes)*:

Changed technologies: **SDK.Wwise**

**2230** apps still match.

These apps now match: **(24)** [1491410](https://steamdb.info/app/1491410/depots/) [1636040](https://steamdb.info/app/1636040/depots/) [1707220](https://steamdb.info/app/1707220/depots/) [1707990](https://steamdb.info/app/1707990/depots/) [1786010](https://steamdb.info/app/1786010/depots/) [1937930](https://steamdb.info/app/1937930/depots/) [1965810](https://steamdb.info/app/1965810/depots/) [2065530](https://steamdb.info/app/2065530/depots/) [2118940](https://steamdb.info/app/2118940/depots/) [2228830](https://steamdb.info/app/2228830/depots/) [2283470](https://steamdb.info/app/2283470/depots/) [2407540](https://steamdb.info/app/2407540/depots/) [2422440](https://steamdb.info/app/2422440/depots/) [3073000](https://steamdb.info/app/3073000/depots/) [3146830](https://steamdb.info/app/3146830/depots/) [3213990](https://steamdb.info/app/3213990/depots/) [3227630](https://steamdb.info/app/3227630/depots/) [3307570](https://steamdb.info/app/3307570/depots/) [3352190](https://steamdb.info/app/3352190/depots/) [3356650](https://steamdb.info/app/3356650/depots/) [3421130](https://steamdb.info/app/3421130/depots/) [3422780](https://steamdb.info/app/3422780/depots/) [3461770](https://steamdb.info/app/3461770/depots/) [3475870](https://steamdb.info/app/3475870/depots/)

Please confirm that all these changed apps are correct.

**Comment by Lifeismana** [comment]
merged in afe87f1485edbe95296172d4bc85fbbb94f49de0
